src/pad_calc.cpp

```diff
@@ -111,5 +122,33 @@ shape compute_padded_shape(const shape& input,
     return input.with_lens(output_lens);
 }
 
+/**
+ *   Calculate the correct output shape for a pooling with
+ *   a given input size and other parameters.  This uses
+ *   the same formula for pooling that compute_padded_shape() uses
+ *   for convolutions, but takes slightly different inputs.
+ *
+ */
+shape compute_padded_pool_shape(const shape& input,
```

Comment:
This and compute_padded_shape() look equivalent. Merge them together and rename the inputs to something that makes sense for both.

Reply:
Better the way it is.  I created this function because the differences in indexing within the shapes' dimension lists made it too messy to reuse compute_padded_shape().